Library/OcBootManagementLib/BootEntryManagement.c

```diff
@@ -1555,14 +1563,40 @@ AddFileSystemEntryForCustom (
       );
   }
 
-  if (BootContext->PickerContext->ShowNvramReset) {
+  if (BootContext->PickerContext->ShowToggleSip) {
+    Status = OcGetSip (&CsrActiveConfig, NULL);
+    if (Status == EFI_NOT_FOUND || !EFI_ERROR(Status)) {
+      LoadOptions = AllocateCopyPool (sizeof (BootContext->PickerContext->ToggleSipMask), &BootContext->PickerContext->ToggleSipMask);
+      if (LoadOptions == NULL) {
+        Status = EFI_OUT_OF_RESOURCES;
+      } else {
+        Status = AddBootEntryFromSystemEntry (
+          BootContext,
+          FileSystem,
+          OcIsSipEnabled (Status, CsrActiveConfig) ? OC_MENU_SIP_IS_ENABLED : OC_MENU_SIP_IS_DISABLED,
```

Comment:
Is this unconditionally initialised regardless of the Status? Sounds like an unusual contract for the function, which I do not like.

Reply:
? I may be confused. The intention is that AddBootEntryFromSystem has not initialise the entry (and has probably printed some warning) if it returns a failure, and if it does the associated LoadOptions memory is then freed just below. I may not be understanding what you are seeing is wrong, yet.

Reply:
You branch from CsrActiveConfig, but it may have returned NOT_FOUND above.

Reply:
Carrying on on NOT_FOUND (and treating that as SIP enabled) is correct, I am almost certain. It is definitely 100% what the Mac does. If there is no such var it (treats it as zero, presumably) and just reports SIP a enabled. So that is what I am trying to do too, here and in lots of places.

Don't forget that with Apple's own CMD+OPT+P+R implementation, this is the state the NVRAM ends up in, so it is definitely not an unsupported corner case.

Reply:
You did not understand my claim. I do agree that the logic is correct, but I do not like the fact the OUT parameter is initialised (overwritten) when the function fails. I think it is EDK-idiomatic not to change the param in such a case. Not a big deal, but a bit strange.

Reply:
Hang on, I'm getting closer to understanding - I think I stayed up too late last night - sorry! Specifically which param is getting initialised even if which function falls? I agree that shouldn't happen, I'm still not seeing where it's happening. Sorry!

Reply:
Status == EFI_NOT_FOUND → OcIsSipEnabled (Status, CsrActiveConfig) is called. I do not think CsrActiveConfig is initialised here.

Reply:
Okay, so, still not sure it is an error. Maybe I'm not using the idioms correctly.

Yes, OcGetSip does NOT initialise CsrActiveConfig on error (including EFI_NOT_FOUND), but the contract of OcIsSipEnabled is in fact to not try to read CsrActiveConfig if EFI_NOT_FOUND is passed in in Status; i.e. it can perfectly well accept an uninitialised param here in that case - in fact expects it to be uninitialised in that case!

So maybe it is alright? I'm not sure.

Reply:
EDK II is not consistent regarding this at all, should not matter.

Reply:
Actually I can see that OcGetSip does initialise CsrActiveConfig, okay - then I need to move that out of there, and do it outside if I need it, right? Okay. Thanks.